File: hyunjae/프로그래머스/고득점키트/기능개발.java
import java.util.*;

class Solution {
    public static int[] solution(int[] progresses, int[] speeds) {
        //정답 담을 리스트 생성
        List<Integer> tempAnswer = new ArrayList<>();
        int n = progresses.length;
        int[] days = new int[n];

        // 각 기능이 완료되기까지 필요한 일수 계산
        for (int i = 0; i < n; i++) {
            int remainingWork = 100 - progresses[i];
            days[i] = (remainingWork + speeds[i] - 1) / speeds[i]; // 올림 처리
        }

        //첫 작업이 끝날 때를 시작 값으로 지정
        int currentDay = days[0];
        //첫 작업 1개가 무조건 들어가므로 1로지정
        int count = 1;

        // 기능 배포 처리
        for (int i = 1; i < n; i++) {
            if (days[i] <= currentDay) {
                count++;
            } else {
                tempAnswer.add(count);
                currentDay = days[i];
                count = 1;
            }
        }
        tempAnswer.add(count);

        // 결과 배열로 변환
        int[] answer = new int[tempAnswer.size()];
        for (int i = 0; i < tempAnswer.size(); i++) {
            answer[i] = tempAnswer.get(i);
        }

        return answer;
    }
}

File: sejeong/기능개발.py
def solution(progresses):
    answer = []
    x = 0
    day = [] #각 작업당 몇일 후 배포해야하는지 담을 배열
    for i in range(len(progresses)):
        if (100-progresses[i])%speeds[i] == 0:
            x = (100-progresses[i])//speeds[i]
        else:
            x = (100-progresses[i])//speeds[i] + 1
        day.append(x)
    stack = [day[0]]
    cnt = 1

    for i in range(1,len(day)):
        if day[i] > stack[-1]: # 뒤에오는 기간이 더길면
            answer.append(cnt) # 현재까지 배포될수 있는 수 담기
            stack[-1] = day[i] # 비교할 수 다시 바꾸기
            cnt = 1 # cnt수 리셋
        else:
            cnt+=1 # 작으면 배포될수없으므로 cnt세기

    answer.append(cnt) # 마지막 cnt수도 넣기
    return answer

progresses = [95, 90, 99, 99, 80, 99]
speeds = [1, 1, 1, 1, 1, 1]
print(solution(progresses))

# 작업에 걸리는 시간이 5,10,1,1,20,1 뒤에올숫자가 작으면 카운트세기